composer.json

```diff
@@ -22,8 +22,25 @@
     "magento-hackathon/magento-composer-installer": "^3.1 || ^2.1 || ^4.0"
   },
   "require-dev": {
+    "friendsofphp/php-cs-fixer": ">=3.4",
     "macopedia/phpstan-magento1": "^1.0.4",
-    "phpstan/phpstan": "^1.8"
+    "magento-ecg/coding-standard": "^4.4",
+    "phpcompatibility/php-compatibility": "^9.3",
+    "phpmd/phpmd": "^2.12",
+    "phpstan/phpstan": "^1.8",
+    "squizlabs/php_codesniffer": "^3.7"
+  },
+  "scripts": {
+    "set-phpcs-paths": [
+      "[ $COMPOSER_DEV_MODE -eq 0 ] || vendor/bin/phpcs --config-set installed_paths vendor/phpcompatibility/php-compatibility,vendor/magento-ecg/coding-standard",
```

Comment:
is there a way not to hardcode vendor?

Reply:
Tried with COMPOSER_VENDOR_DIR, but it does not work.

IMHO ... having a custom vendor dir is an edge case. If this command fails for someone, i'm pretty sure they know how to fix.

BTW ... seems COMPOSER_VENDOR_DIR has to be set as ENV variable, but does no take the value from composer config section. (@Flyingmana you`re familiar with composer ... is this right and/or intended behavior?)

Reply:
Tried with COMPOSER_VENDOR_DIR ....

... if you set COMPOSER_VENDOR_DIR as ENV variable it works ...

ddev config.yaml:

```
web_environment: [
  COMPOSER_VENDOR_DIR=test
]
```

composer.json
```
    "set-phpcs-paths": [
      "[ $COMPOSER_DEV_MODE -eq 0 ] || $COMPOSER_VENDOR_DIR/bin/phpcs --config-set installed_paths $COMPOSER_VENDOR_DIR/phpcompatibility/php-compatibility,$COMPOSER_VENDOR_DIR/magento-ecg/coding-standard",
      "[ $COMPOSER_DEV_MODE -eq 0 ] || $COMPOSER_VENDOR_DIR/bin/phpcs -i"
    ],
```

Having custom vendor dir is an edge case ... i'd not change it.